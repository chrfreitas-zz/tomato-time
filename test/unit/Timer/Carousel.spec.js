import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'

import Carousel from '../../../src/modules/timer/components/carousel/carousel.component';

const localVue = createLocalVue()
localVue.use(Vuex)

describe('Timer:Carousel', () => {
  let store
  let actions

  beforeEach(() => {
    actions = {
      changeTo: jest.fn()
    }

    const timer = {
      getters: {
        isDisabledNext: jest.fn(),
        isDisabledPrevious: jest.fn()
      },
      actions,
    }
    store = new Vuex.Store({
      modules: {
        timer,
      }
    })
  })

  it('should create/match snapshot', () => {
    const wrapper = shallowMount(Carousel, {store, localVue});
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should stop the timer', () => {
    const wrapper = shallowMount(Carousel, {store, localVue});
    wrapper.find('button:first-child').trigger('click');
    expect(actions.changeTo.mock.calls).toHaveLength(1)
  });
});
