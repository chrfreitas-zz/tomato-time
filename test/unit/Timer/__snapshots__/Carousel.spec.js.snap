// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer:Carousel should create/match snapshot 1`] = `
<div class="carousel">
    <button class="btn">
        <i class="fas fa-angle-left"></i>
    </button>
    <button class="btn">
        <i class="fas fa-angle-right"></i>
    </button>
</div>
`;
